src/main/scala/is/hail/expr/Type.scala

```diff
@@ -1469,6 +1473,107 @@ final case class TStruct(fields: IndexedSeq[Field]) extends Type {
     filter(fn)
   }
 
+  def ungroup(identifier: String, mangle: Boolean = false): (TStruct, (Row) => Row) = {
+    val overlappingNames = mutable.Set[String]()
+
+    val ungroupedFields = fieldOption(identifier) match {
+      case None => fatal(s"Struct does not have field with name `$identifier'.")
+      case Some(x) => x.typ match {
+        case s: TStruct =>
+          s.fields.flatMap { fd =>
+            (fieldNames.contains(fd.name), mangle) match {
+              case (_, true) => Some((identifier + "." + fd.name, fd.typ))
+              case (false, false) => Some((fd.name, fd.typ))
+              case (true, false) =>
+                overlappingNames += fd.name
+                None
+            }
+          }.toArray
+        case other => fatal(s"Expecting a type of TStruct for field `$identifier', but found ${ other.toPrettyString() } ")
```

Comment:
needs to be `compact=True`. I think the message is also a bit unclear, and the stuff in the Python docs uses `Struct` instead of `TStruct`. How about:

"Can only ungroup fields of type Struct, but found type ${...}"